__tests__/integration.test.js

```diff
@@ -104,4 +104,43 @@ describe('GET /api/articles', () => {
             }
         })
     });
+});
+describe('GET /api/articles/:article_id/comments', () => {
+    test('status 200: should return array of comments for the given article_id, each comment with the following properties: comment_id, votes, created_at, author, body, article_id, ordered by date descending', () => {
+        return request(app)
+        .get("/api/articles/9/comments")
+        .expect(200)
+        .then(({body}) => {
+            const comments = body.comments
+            expect(comments).toHaveLength(2)
+            expect(comments).toBeSortedBy('created_at',{descending: true})
+            comments.forEach((comment) => {
+                expect(comment).toMatchObject({
+                    comment_id: expect.any(Number),
+                    votes: expect.any(Number),
+                    created_at: expect.any(String),
+                    author: expect.any(String),
+                    body: expect.any(String),
+                    article_id: expect.any(Number)
+                })
+            })
+        })
+    });
+    test('status 200: should return empty array when given an article_id with no comments', () => {
+        return request(app)
+        .get("/api/articles/2/comments")
+        .expect(200)
+        .then(({body}) => {
+            const comments = body.comments
+            expect(comments).toHaveLength(0)
+        })
+    });
+    test('status 404: should return error message when given an article_id which doesnt exist in the articles table', () => {
+        return request(app)
+        .get("/api/articles/999999999/comments")
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe("Article Not Found")
+        })
+    });
```

Comment:
400 invalid input ("api/articles/banana/comments") is also needed here

Reply:
of course!!! i have added this into my current branch (08-patch-article-by-article-id) 😄 thank so much

Reply:
i am also adding this in for 07 as i now realise i missed it from there as well, thanks so much for pointing this out, a very important step! 🙌